elements/dash-accordion/dash-accordion.js

```diff
@@ -0,0 +1,529 @@
+
+      // If this element doesn’t have an id, generate one.
+      if (!this.id)
+        this.id =
+          `dash-accordion-heading-generated-${dashAccordionHeadingCounter++}`;
+    }
+
+    /**
+     * `isExpanded` is a getter to query if the associated panel is expanded
+     * or collpased.
+     */
+    get isExpanded() {
+      return this.getAttribute('aria-expanded') === 'true';
+    }
+
+    /**
+     * `_displayToggleEvent` emits either a `accordion-collapse` or an
+     * `accordion-expand` event depending on the current state.
+     */
+    _dispatchToggleEvent() {
+      if (this.isExpanded)
+        this.dispatchEvent(
+          new CustomEvent('accordion-collapse', {bubbles: true})
+        );
+      else
+        this.dispatchEvent(
+          new CustomEvent('accordion-expand', {bubbles: true})
+        );
+    }
+
+    /**
+     * Event handle for the `keydown` event.
+     */
+    _onKeyDown(event) {
+      // Don’t handle modifier shortcuts typically used by assistive technology.
+      if (event.altKey) return;
+
+      switch (event.keyCode) {
+        case KEYCODE.SPACE:
+        case KEYCODE.ENTER:
+          this._dispatchToggleEvent();
+          break;
+        case KEYCODE.LEFT:
+        case KEYCODE.UP:
+          this.dispatchEvent(
+            new CustomEvent('accordion-previous', {bubbles: true})
+          );
+          break;
+
+        case KEYCODE.RIGHT:
+        case KEYCODE.DOWN:
+          this.dispatchEvent(
+            new CustomEvent('accordion-next', {bubbles: true})
+          );
+          break;
+
+        case KEYCODE.HOME:
+          this.dispatchEvent(
+            new CustomEvent('accordion-first', {bubbles: true})
+          );
+          break;
+
+        case KEYCODE.END:
+          this.dispatchEvent(
+            new CustomEvent('accordion-last', {bubbles: true})
+          );
+          break;
+        // Any other key press is ignored and passed back to the browser.
+        default:
+          return;
+      }
+
+      // The browser might have some native functionality bound to the arrow
+      // keys, home or end. The element calls `preventDefault` to prevent the
+      // browser from taking any actions.
+      event.preventDefault();
+    }
+
+    /**
+     * Event handler for `click` event.
+     */
+    _onClick() {
+      this._dispatchToggleEvent();
+    }
+  }
+  window.customElements.define('dash-accordion-heading', DashAccordionHeading);
+
+  let dashAccordionPanelCounter = 0;
+  /**
+   * `DashAccordionPanel` is a panel for a `<dash-accordion>`.
+   */
+  class DashAccordionPanel extends HTMLElement {
+    constructor() {
+      super();
+    }
+
+    connectedCallback() {
+      this.setAttribute('role', 'region');
+      this.setAttribute('aria-hidden', 'true');
+      if (!this.id)
+        this.id =
+          `dash-accordion-panel-generated-${dashAccordionPanelCounter++}`;
+    }
+  }
+  window.customElements.define('dash-accordion-panel', DashAccordionPanel);
```

Comment:
I think with tabs you went with `dash-tabpanel`. Do you think it would be worthwhile to go back and kebab case tabpanel to match accordion-panel?